examples/custom-assessment/src/main.py

```diff
@@ -0,0 +1,199 @@
+we cannot use ray.put() to pass them as arguments to remote functions.
+Instead, we will use RLlib’s checkpointing mechanism to save and load the
+Algorithm state for each worker.
+"""
+import argparse
+import json
+import os
+import sys
+from pathlib import Path
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+import ray
+from ray.rllib.algorithms.algorithm import Algorithm
+from ray.tune.registry import register_env
+from ray_on_aml.core import Ray_On_AML
+from sim import (
+    SimpleAdder as SimEnv,  # IMPORTANT: Change to import your own environment
+)
+
+# Disable deduplication of Ray logs.
+os.environ["RAY_DEDUP_LOGS"] = "0"
+
+# IMPORTANT: sim_env must be equal to the name used to register the environment when
+# training the agent.
+register_env("sim_env", lambda conf: SimEnv(conf))
+
+
+@ray.remote
+class EpisodeRunner:
+    """Class to run an episode of the simulation using a trained agent's policy."""
+
+    def __init__(
+        self,
+        checkpoint_folder: str,
+    ):
+        """Initialize the episode runner by instantiating the agent.
+
+        Args:
+            checkpoint_folder (str): The name of the folder where the checkpoint is
+            located.
+        """
+        # Instantiate the trained agent from the checkpoint folder
+        checkpoint_path = Path(__file__).absolute().parent.parent / checkpoint_folder
+        self.agent = Algorithm.from_checkpoint(checkpoint_path)
+
+        # Initialize the simulation environment
+        # IMPORTANT: Change to import your own environment
+        self.sim = SimEnv(env_config={})
+
+    def episode_to_df(
+        self, states, actions, rewards, terminated, truncated, iterations
+    ) -> pd.DataFrame:
+        """Create Pandas dataframe from episode iterations.
+
+        Args:
+            states (dict): The states of the simulation at each iteration.
+            actions (dict): The actions of the agent at each iteration.
+            rewards (list): The rewards of the agent at each iteration.
+            terminated (list): Whether the episode terminated at each iteration.
+            truncated (list): Whether the episode was truncated at each iteration.
+            iterations (int): The final number of iterations in the episode.
+
+        Returns:
+            pd.DataFrame: The dataframe containing the states, actions, episode id,
+            iteration number, and initial condition.
+        """
+        df = pd.concat([pd.DataFrame(dict) for dict in [states, actions]], axis=1)
+        for key, value in self.initial_condition.items():
+            df["config_" + key] = value
+        df["reward"] = pd.Series(rewards)
+        df["reward_cumsum"] = df["reward"].cumsum()
+        df["terminated"] = pd.Series(terminated)
+        df["truncated"] = pd.Series(truncated)
+        df["episode"] = self.episode_id
+        df["iteration"] = pd.Series(np.arange(iterations + 1))
```

Comment:
I think it's better to make this an ``int``.

Reply:
Oh yes, Pandas' series is fine, I was just wondering if you could make this an integer series rather than a float. If I look at the final csv, I get iteration for instance `0.0`. Just wanted to ask if you could make this ``0`` instead.